src/librustdoc/clean/mod.rs

```diff
@@ -1560,7 +1560,7 @@ impl Clean<Generics> for hir::Generics {
 }
 
 impl<'a, 'tcx> Clean<Generics> for (&'a ty::Generics,
-                                    &'a ty::GenericPredicates<'tcx>) {
+                                    &'a Rc<ty::GenericPredicates<'tcx>>) {
```

Comment:
Not using `Lrc` here is probably wrong, but why add `Rc` at all? If inference doesn't work, you can just add a `*` in the caller. (or are there many callers?)

Reply:
Oh, it is necessary. Unless I change the callsites as you suggest, but there are 11 of those.